src/Symfony/Component/Security/Core/Encoder/MigratingPasswordEncoder.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Core\Encoder;
+
+/**
+ * Hashes passwords using the best available encoder.
+ * Validates them using a chain of encoders.
+ *
+ * /!\ Don't put a PlaintextPasswordEncoder in the list as that'd mean a leaked hash
```

Comment:
why not looping inside the __construct on encoders and throw ex if PlaintextPasswordEncoder is found?
I do not think developers are going to read this :/

Reply:
Because we don't do politics :)
If ppl want to do it, their choice. Also because hardcoding a class name wouldn't prevent anyone from implementing this under a different name and be exposed.

Reply:
understood :)